ankihub/gui/decks_dialog.py

```diff
@@ -911,22 +919,89 @@ def _on_new_cards_destination_btn_clicked(self):
         if current_destination_deck := get_deck_for_ah_did(ah_did):
             current_destination_deck_name = current_destination_deck["name"]
 
-        def update_deck_config(note_type_selector: SearchableSelectionDialog):
+        def on_destination_selected(note_type_selector: SearchableSelectionDialog):
             if not note_type_selector.name:
                 return
 
-            anki_did = aqt.mw.col.decks.id(note_type_selector.name)
+            new_deck_name = note_type_selector.name
+
+            should_move_cards = False
+            if (
+                not current_destination_deck_name
+                or new_deck_name != current_destination_deck_name
+            ):
+                anki_nids = ankihub_db.anki_nids_for_ankihub_deck(ah_did)
+                if not all_notes_in_deck(
+                    nids=anki_nids, anki_did=aqt.mw.col.decks.id(new_deck_name)
+                ):
+                    should_move_cards = self._show_move_cards_dialog(new_deck_name)
+
+            # Update the destination deck configuration
+            anki_did = aqt.mw.col.decks.id(new_deck_name)
```

Comment:
:pick: to be more clear, maybe we can change this variable name from `anki_did` to `new_destination_anki_did` or something similar